networkx/algorithms/time_dependent.py

```diff
@@ -0,0 +1,118 @@
+"""Time dependent algorithms."""
+
+from datetime import datetime
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+__all__ = ["cd_index"]
+
+
+@not_implemented_for("undirected")
+@not_implemented_for("multigraph")
+def cd_index(G, node, time_delta=5, weight=None):
+    r"""Compute the CD index.
+
+    Calculates the CD index for the graph based on the given "focal patent" node
+    considering the patents after `time_delta` years.
+
+    Parameters
+    ----------
+    G : graph
+       A directed networkx graph whose nodes have datetime `time` attributes.
+    node : integer
```

Comment:
Nodes are not integers in general.  They can be any hashable. So, you can use `node : node`  just as you do with `G : graph`.